src/Gateway/PayplugCreditCard.php

```diff
@@ -248,4 +249,113 @@ public function payment_fields()
 			$this->saved_payment_methods();
 		}
 	}
+
+	/**
+	 * Process the subscription scheduled payment
+	 */
+	public function scheduled_subscription_payment($amount, $order) {
+
+		$order_id      = PayplugWoocommerceHelper::is_pre_30() ? $order->id : $order->get_id();
+		$customer_id = PayplugWoocommerceHelper::is_pre_30() ? $order->customer_user : $order->get_customer_id();
+
+		$subscription = wcs_get_subscription($order->get_meta('_subscription_renewal'));
+
+		$payplug_meta = $subscription->get_parent()->get_meta("_payplug_metadata");
```

Comment:
this should be renamed, (parent_data, subscription, ... )

Reply:
fixed + adding a fail safe